packages/core/helper-plugin/src/utils/request.ts

```diff
@@ -92,30 +85,26 @@ function serverRestartWatcher(response) {
 
 const warnOnce = once(console.warn);
 
+interface RequestOptions extends RequestInit {
+  params?: Record<string, string>;
+}
+
 /**
  * Requests a URL, returning a promise
  *
  * @deprecated use `useFetchClient` instead.
- *
- * @param  {string} url       The URL we want to request
- * @param  {object} [options] The options we want to pass to "fetch"
- *
- * @return {object}           The response data
  */
-export function request(...args) {
-  let [url, options = {}, shouldWatchServerRestart, stringify = true, ...rest] = args;
-  let noAuth;
-
+export async function request<ResponseType = unknown>(
+  url: string,
+  options = {} as RequestOptions,
```

Comment:
why the type casting?

```suggestion
  options: RequestOptions = {},
```

Reply:
you're right